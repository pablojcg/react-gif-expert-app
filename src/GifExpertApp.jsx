
import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Real Madrid', 'New York Yankees', 'Apple']);

    const onAddCategory = (newCategory) => {
        // Validar si la categoria ya existe
        if (categories.includes(newCategory)) return;
        setCategories([newCategory, ...categories]);
    };

    return (
        <>
            <h1>GifExpertApp PC</h1>

            <AddCategory onAddCategory={onAddCategory} />

            {
                categories.map(category => (
                    <GifGrid key={category} category={category} />
                ))
            }
        </>
    );
}