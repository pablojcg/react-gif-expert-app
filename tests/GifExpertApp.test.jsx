import { render } from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';

describe('Probando Componente <GifExpertApp />', () => {
    
    const title = 'GifExpertApp PC';

    test('Debe hacer match con el snapshot', () => {
        const { container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar el titulo "GifExpertApp PC"', () => {
        const { getByText } = render(<GifExpertApp />);
        expect(getByText(title)).toBeTruthy();
    });

    test('Debe mostrar el titulo "GifExpertApp PC" en un h1', () => {
        const { getByRole } = render(<GifExpertApp />);
        expect(getByRole('heading', { level: 1 }).innerHTML).toBe(title);
    });
});