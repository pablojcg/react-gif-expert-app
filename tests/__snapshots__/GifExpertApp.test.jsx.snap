// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Probando Componente <GifExpertApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp PC
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    Real Madrid
  </h3>
  <h2>
    Cargando...
  </h2>
  <h3>
    New York Yankees
  </h3>
  <h2>
    Cargando...
  </h2>
  <h3>
    Apple
  </h3>
  <h2>
    Cargando...
  </h2>
</div>
`;
